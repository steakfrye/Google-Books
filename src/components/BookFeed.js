import React, { Component } from 'react';
import BookItem from './BookItem';

class BookFeed extends Component {
  render() {
    const { books, showModal } = this.props;

    return (
      books.map(book => <BookItem key={book.id} book={book} showModal={showModal}/>)
    );
  }
}

export default BookFeed;
