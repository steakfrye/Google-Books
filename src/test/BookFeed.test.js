import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { configure, mount, shallow } from 'enzyme';
import BookFeed from '../components/BookFeed';

configure({ adapter: new Adapter() });

describe('<BookFeed />', () => {
  const books = [{
    "id": "bZ_JqLD6NAEC",
    "volumeInfo": {
      "title": "Collaborative Quilting",
      "authors": [
       "Freddy Moran",
       "Gwen Marston"
     ],"publisher": "Sterling Publishing Company, Inc.",
     "imageLinks": {"thumbnail": "http://tinyurl.com/55"}
   }
 },
 {
   "id": "bZ_JqLD6NADC",
   "volumeInfo": {
     "title": "Collaborative Pizza",
     "authors": [
      "James Morgan",
      "Greg Mangus"
    ],"publisher": "The Publishing Company, Inc.",
    "imageLinks": {"thumbnail": "http://tinyurl.com/54"}
  }
 }]
  const item = mount(<BookFeed books={books}/>)

  it('renders data from server', () => {
  const tree = renderer
    .create(item)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

  it('renders book info from props', () => {
    expect(item.contains(<td>Freddy Moran, Gwen Marston</td>)).toBe(true);
    expect(item.contains(<td>Collaborative Quilting</td>)).toBe(true);
    expect(item.contains(<td>Sterling Publishing Company, Inc.</td>)).toBe(true);
  });
});
