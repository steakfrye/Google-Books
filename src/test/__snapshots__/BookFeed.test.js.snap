// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookFeed /> renders data from server 1`] = `
Array [
  <tr>
    <td>
      <img
        alt="Collaborative Quilting"
        src="http://tinyurl.com/55"
      />
    </td>
    <td>
      Collaborative Quilting
    </td>
    <td>
      Freddy Moran, Gwen Marston
    </td>
    <td>
      Sterling Publishing Company, Inc.
    </td>
  </tr>,
  <tr>
    <td>
      <img
        alt="Collaborative Pizza"
        src="http://tinyurl.com/54"
      />
    </td>
    <td>
      Collaborative Pizza
    </td>
    <td>
      James Morgan, Greg Mangus
    </td>
    <td>
      The Publishing Company, Inc.
    </td>
  </tr>,
]
`;
